packages/manager/apps/web/client/app/domain/zone/activate/activate.controller.js

```diff
@@ -12,20 +13,29 @@ export default class DomainDnsZoneActivateController {
     $window,
     Alerter,
     DomainDnsZoneActivateService,
+    ovhManagerProductOffersService,
   ) {
     this.$filter = $filter;
     this.$timeout = $timeout;
     this.$translate = $translate;
     this.$window = $window;
     this.Alerter = Alerter;
     this.DomainDnsZoneActivateService = DomainDnsZoneActivateService;
+    this.ovhManagerProductOffersService = ovhManagerProductOffersService;
   }
 
   $onInit() {
     // Auto-select duration
-    this.prices = filter(
-      this.serviceOption.prices,
-      (price) => price.pricingMode === pricingConstants.PRICING_MODE.DEFAULT,
+    this.prices = uniqBy(
+      filter(
+        this.serviceOption.prices,
+        (price) =>
+          [
+            pricingConstants.PRICING_MODE.DEFAULTS.CREATE,
+            pricingConstants.PRICING_MODE.DEFAULTS.DEFAULT,
+          ].includes(price.pricingMode) && price.interval !== 0,
```

Comment:
Ugly solution, we should be able to not filter many thing to have correct pricings, but this requires a drastic change. (Use of other routes)